### ISSUE TITLE
Git-files should support {{path}}

### ISSUE THREAD

**Issue by paveq** [argoproj-labs/applicationset#258]
It would be useful to be able to use {{path}} template variable in git-files generator, to get path of config.json in thee source repo. We have config.json in the same directory as manifests, and it is little bit silly having to those path to config.json within the config.json file itself.

**Comment by jgwest** [comment]
Agreed @paveq, I presume you are looking for the **folder path** that the `config.json` files is in (for example,  `/clusters/clusterA`) rather than a path to the actual config file (for example, `/clusters/clusterA/config.json`)?

**Comment by bitfactory-henno-schooljan** [comment]
This would be a great addition. I currently have configurations in the form  of `/application/environment.yaml`, e.g. `/app1/test.yml`, `/app2/acceptance.yaml` and so on, and it would be nice if I could extract these components from the path somehow instead of having to duplicate these into the configuration files as variables.

With the folder path I could use the `{{ path.basename }}` to extract the application name in this case, not sure how I would extract multiple components as I cannot simply use Sprig style functions here.

**Comment by jgwest** [comment]
So then the new parameters would be:
- `{{ path }}` = path to the folder containing the config.json (eg `/clusters/clusterA`) 
- `{{ path.basename }}` = basename of path to the folder (eg `clusterA`)
